components/validator/guardian.go

```diff
@@ -533,6 +533,11 @@ func (g *Guardian) shouldBeDeleted(outputIndex, fromBlock, toBlock *big.Int) (bo
 		return false, fmt.Errorf("failed to get output from L2OutputOracle contract(outputIndex: %d): %w", outputIndex.Uint64(), err)
 	}
 
+	if IsOutputDeleted(output.OutputRoot) {
```

Comment:
This seems like a similar property to conditions like isFinalized above the call. Why not put all of this conditional checking logic in the same place at the same level? (guardian.go:280)

Reply:
The function `IsOutputDeleted` is in `challenger.go`.
As you mentioned, these kinds of function is spread all over the validator.
I thought that we are gonna make refactoring for future and arrange them at that time.